packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -207,10 +211,47 @@ export class KubernetesClient {
   }
 
   getCurrentNamespace(): string | undefined {
-    return this.currrentNamespace;
+    return this.currentNamespace;
+  }
+
+  private async getDefaultNamespace(context: Context): Promise<string> {
+    if (context.namespace) {
+      return context.namespace;
+    }
+    const ctx = new KubeConfig();
+    ctx.loadFromOptions({
+      currentContext: context.name,
+      clusters: this.kubeConfig.clusters,
+      contexts: this.kubeConfig.contexts,
+      users: this.kubeConfig.users,
+    });
+    let namespace;
+
+    try {
+      const cm = await this.readNamespacedConfigMap(OPENSHIFT_CONSOLE_CONFIG_MAP, OPENSHIFT_CONSOLE_NAMESPACE);
+      if (cm) {
+        const projects = await ctx
+          .makeApiClient(CustomObjectsApi)
+          .listClusterCustomObject('project.openshift.io', 'v1', 'projects');
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        if ((projects?.body as any)?.items.length > 0) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          namespace = (projects?.body as any)?.items[0].metadata?.name;
+        }
+      }
+    } catch (err) {
+      const namespaces = await ctx.makeApiClient(CoreV1Api).listNamespace();
+      if (namespaces?.body?.items.length > 0) {
+        namespace = namespaces?.body?.items[0].metadata?.name;
+      }
+    }
+    if (!namespace) {
+      namespace = 'default';
+    }
+    return namespace;
   }
 
-  refresh() {
+  async refresh() {
```

Comment:
No because this function is called to trigger an update to the context

Reply:
I mean, it's a pattern.
it may lead to unhandled errors (let say tomorrow someone include in refresh something that is causing an error

https://eslint.org/docs/latest/rules/require-await